incubator/hnc/pkg/reconcilers/config_reconciler.go

```diff
@@ -1,4 +1,4 @@
-package controllers
+package reconcilers
```

Comment:
Maybe change the filename too? `reconcilers/config_reconciler.go` seems both redundant and a bit ambiguous because of HierarchyConfiguration. How about `reconcilers/hncconfig.go` (and also `object.go` and `hierarchyconfig.go`)?

Reply:
Changed to hnc_config.go and hierarchy_config.go since there are two words. Changed object_reconciler.go to object.go. Feel free to take a look.

Reply:
Hmm - seems like the convention is _not_ to use underscores? See https://stackoverflow.com/questions/25161774/what-are-conventions-for-filenames-in-go. Have we not been following this?

Reply:
Hmm, looks like we don't follow that. lgtm :)